deploy/docker/scripts/supervisor_event_listener.py

```diff
@@ -0,0 +1,74 @@
+import os
+import requests
+import sys
+import shutil
+import time
+
+LOADING_TEMPLATE_PAGE = r'/opt/appsmith/templates/appsmith_starting.html'
+LOADING_PAGE_EDITOR = r'/opt/appsmith/editor/loading.html'
+BACKEND_HEALTH_ENDPOINT = "http://localhost/api/v1/health"
+
+def write_stdout(s):
+    # only eventlistener protocol messages may be sent to stdout
+    sys.stdout.write(s)
+    sys.stdout.flush()
+
+def write_stderr(s):
+    sys.stderr.write(s)
+    sys.stderr.flush()
+
+def wait_until_backend_healthy():
+    sleep_sec = 3
+    timeout_sec = 45
+    for _ in range(timeout_sec//sleep_sec):
+        if requests.get(BACKEND_HEALTH_ENDPOINT).ok:
+            write_stderr('\nBackend is healthy\n')
+            break
+        time.sleep(sleep_sec)
+
+    else:
+        write_stderr('\nBackend health check timed out\n')
+
+    remove_loading_page()
+
+def remove_loading_page():
+    if os.path.exists(LOADING_PAGE_EDITOR):
+        os.remove(LOADING_PAGE_EDITOR)
+
+def main():
+    while 1:
+        # transition from ACKNOWLEDGED to READY
+        write_stdout('READY\n')
+
+        # read header line and print it to stderr
+        line = sys.stdin.readline()
+        write_stderr(line)
+
+        # read event payload and print it to stderr
+        headers = dict([ x.split(':') for x in line.split() ])
```

Comment:
Ensure that the split is done exactly _once_. If the message contains `:` characters, we don't want it to split further, as the `dict` constructor expects 2-tuplis in the iterator.

We also don't need the `[]`, since `dict` can take a generator expression.

There's a few other places we can make this change too.

```suggestion
        headers = dict(x.split(':', 1) for x in line.split())
```